### PR TITLE
[credentialhelper] Add types to communicate with the subprocess

### PR DESCRIPTION
**Pull request by Yannic**
This change adds `GetCredentials{Request,Response}`, which we'll use to
pass data in `stdin` to the credental helper subprocess, and read the
response from `stdout` of the subprocess.

Progress on https://github.com/bazelbuild/proposals/blob/main/designs/2022-06-07-bazel-credential-helpers.md#proposal

### CODE REVIEW

**Comment by Yannic** [comment]
@tjgq PTAL

File: src/test/java/com/google/devtools/build/lib/authandtls/credentialhelper/GetCredentialsResponseTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    expectedResponseBuilder.headersBuilder().put("c", ImmutableList.of("c", "c"));
+    GetCredentialsResponse expectedResponse = expectedResponseBuilder.build();
+
+    assertThat(
+            GSON.fromJson(
+                "{\"headers\": {\"c\": [\"c\", \"c\"], \"a\": [], \"b\": [\"b\"]}}",
+                GetCredentialsResponse.class))
+        .isEqualTo(expectedResponse);
+  }
+
```
Comment (tjgq) [inline src/test/java/com/google/devtools/build/lib/authandtls/credentialhelper/GetCredentialsResponseTest.java:34]: Can you add tests for malformed `headers` fields, e.g.

`{"headers": null}`
`{"headers": "foo"}`
`{"headers": ["foo"]}`
`{"headers": {"foo": "bar"}}`

**Review by tjgq — CHANGES_REQUESTED** [review]

**Review by tjgq — APPROVED** [review]

**Comment by Yannic** [comment]
@bazel-io flag

**Comment by ckolli5** [comment]
@bazel-io fork 5.3.0

**Pull request closed without merging**